scripts/nbconverted/predictive_genes.py

```diff
@@ -93,27 +108,78 @@
 Y = pd.merge(estimated_latent_df, obsv_df, left_index=True, right_index=True)
 
 # Residuals: observed - estimated
-Y['residuals'] = X['observed']-X['estimated']
+Y['residuals_original'] = abs(Y['observed']-Y['estimated'])
 Y.head(5)
 
 # Mean and stdev of residuals
-residual_mean = Y['residuals'].values.mean()
-residual_std = Y['residuals'].values.std()
+#residual_mean = Y['residuals'].values.mean()
+#residual_std = Y['residuals'].values.std()
 
-print("Mean of residuals is: {} \n Standard deviation of residuals is: {}".format(residual_mean, residual_std))
+#print("Mean of residuals is: {} \n Standard deviation of residuals is: {}".format(residual_mean, residual_std))
 
 # Identify those genes that have a residual that are exceed 3 std from mean 
 # Genes that are not predictive using linear space
-threshold = residual_std*3
-linear_notPredictive_geneID = X.index[X['residuals']>= threshold].tolist()
+#threshold = residual_std*3
+#linear_notPredictive_geneID = X.index[X['residuals']>= threshold].tolist()
 #linear_notPredictive_geneID
 
 
-# In[8]:
+# In[9]:
+
+
+# Join 
+Z = pd.merge(X, Y, left_index=True, right_index=True)
+Z.head(5)
+
+
+# In[10]:
+
+
+# Threshold
+# Set threshold based on the dropoff after the shoulder from manual visual inspection
+latent_threshold = 0.03
+original_threshold = 0.03
+
+
+# In[11]:
+
+
+# Plot
+fg=sns.jointplot(x='residuals_original', y='residuals_latent', data=Z, kind='hex');
+fg.ax_joint.axhline(y=latent_threshold, color='k', linestyle='--')
+fg.ax_joint.axvline(x=original_threshold, color='k', linestyle='--',)
+fg.savefig(fig_file)
+
+
+# In[12]:
 
 
 # Intersection of lists
 # Genes that are predictive in latent space and NOT predictive in linear space
-target_geneID = list(set(latent_predictive_geneID) & set(linear_notPredictive_geneID))
-print("Number of target genes is {}".format(len(target_geneID)))
+
+latent_predictive = Z.index[Z['residuals_latent'] < latent_threshold].tolist()
```

Comment:
I am not sure what this is doing. What does it mean to be predictive in the latent space?

Reply:
A gene is considered "predictive" is one where the correlation of the predicted and the actual gene expression is high.  This script is attempting to determine if there were certain genes where the predicted expression using gene space is high compared to the actual expression.  Same for the latent space prediction.  

I have added some comments to the header of the script to describe this more.  But will definitely do more cleaning in the refactoring